lib/vdc/BOVCollection.cpp

```diff
@@ -298,21 +354,22 @@ std::array<double, 3> BOVCollection::GetBrickOrigin() const { return _brickOrigi
 
 std::array<double, 3> BOVCollection::GetBrickSize() const { return _brickSize; }
 
-std::string BOVCollection::GetDataEndian() const { return _dataEndian; }
-
 // Template specialization for reading data of type DC::XType
 template<> int BOVCollection::_findToken<DC::XType>(const std::string &token, std::string &line, DC::XType &value, bool verbose)
 {
     // Skip comments
     for (size_t i = 0; i < line.length(); i++) {
         if (line[i] == '#') {
             line.erase(line.begin() + i, line.end());
-            if (line[line.length() - 1] == ' ')    // If last char is a space, pop it
-                line.pop_back();
             break;
         }
     }
 
+    if (line.length() == 0) return (int)parseCodes::NOT_FOUND;
+
+    while (line[line.length() - 1] == ' ')    // If last char is a space, pop it
```

Comment:
Won't this generate an invalid memory access if line contains only a single space?

Reply:
Fixed.